File: app/src/main/java/vn/loitp/app/activity/customviews/imageview/scrollparallaximageview/ScrollParallaxImageViewActivity.java
package vn.loitp.app.activity.customviews.imageview.scrollparallaximageview;

import android.os.Bundle;
import android.widget.LinearLayout;

import com.core.base.BaseFontActivity;
import com.views.imageview.scrollparallax.LScrollParallaxImageView;
import com.views.imageview.scrollparallax.parallaxstyle.HorizontalScaleStyle;
import com.views.imageview.scrollparallax.parallaxstyle.VerticalMovingStyle;

import loitp.basemaster.R;

//https://github.com/gjiazhe/ScrollParallaxImageView?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=4759
public class ScrollParallaxImageViewActivity extends BaseFontActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        LScrollParallaxImageView iv = findViewById(R.id.img);
        iv.setParallaxStyles(new VerticalMovingStyle()); // or other parallax styles

        LinearLayout llHorizontal = findViewById(R.id.ll_horizontal);
        for (int i = 0; i < 10; i++) {
            LScrollParallaxImageView LScrollParallaxImageView = new LScrollParallaxImageView(getActivity());
            LScrollParallaxImageView.setImageResource(i % 2 == 0 ? R.drawable.iv : R.drawable.logo);
            LScrollParallaxImageView.setParallaxStyles(new HorizontalScaleStyle()); // or other parallax styles
            llHorizontal.addView(LScrollParallaxImageView);
        }

        LinearLayout ll = findViewById(R.id.ll);
        for (int i = 0; i < 20; i++) {
            LScrollParallaxImageView LScrollParallaxImageView = new LScrollParallaxImageView(getActivity());
            LScrollParallaxImageView.setImageResource(i % 2 == 0 ? R.drawable.iv : R.drawable.logo);
            LScrollParallaxImageView.setParallaxStyles(new VerticalMovingStyle()); // or other parallax styles
            ll.addView(LScrollParallaxImageView);
        }
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_scrollparallax_imageview;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/imageview/scrollparallaximageview/ScrollParallaxImageViewActivity.kt
package vn.loitp.app.activity.customviews.imageview.scrollparallaximageview

import android.os.Bundle
import com.annotation.IsFullScreen
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.views.imageview.scrollparallax.LScrollParallaxImageView
import com.views.imageview.scrollparallax.parallaxstyle.HorizontalScaleStyle
import com.views.imageview.scrollparallax.parallaxstyle.VerticalMovingStyle
import kotlinx.android.synthetic.main.activity_imageview_scrollparallax.*
import vn.loitp.app.R

// https://github.com/gjiazhe/ScrollParallaxImageView

@LogTag("ScrollParallaxImageViewActivity")
@IsFullScreen(false)
class ScrollParallaxImageViewActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_imageview_scrollparallax
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        spiv.setParallaxStyles(VerticalMovingStyle()) // or other parallax styles
        for (i in 0..9) {
            val lScrollParallaxImageView = LScrollParallaxImageView(this)
            lScrollParallaxImageView.setImageResource(if (i % 2 == 0) R.drawable.iv else R.drawable.logo)
            lScrollParallaxImageView.setParallaxStyles(HorizontalScaleStyle()) // or other parallax styles
            llHorizontal.addView(lScrollParallaxImageView)
        }
        for (i in 0..19) {
            val lScrollParallaxImageView = LScrollParallaxImageView(this)
            lScrollParallaxImageView.setImageResource(if (i % 2 == 0) R.drawable.iv else R.drawable.logo)
            lScrollParallaxImageView.setParallaxStyles(VerticalMovingStyle()) // or other parallax styles
            ll.addView(lScrollParallaxImageView)
        }
    }
}
